### ISSUE TITLE
partial_dependency failing when provided an explainer object with model.matrix data

### ISSUE THREAD

**Issue by demirev** [ModelOriented/ingredients#26]
Reproducing the example for using DALEX with xgboost found here: https://pbiecek.github.io/DALEX/articles/DALEX_and_xgboost.html but replacing `single_variable` with `ingredients::partial_dependency()` (as the `single_variable` function itself throws a deprecation warning) produces an error.

```
library(dplyr)
library(xgboost)
library(DALEX)
library(ingredients)

wine <- breakDown::wine

model_matrix_train <- model.matrix(quality ~ . - 1, wine)
data_train <- xgb.DMatrix(model_matrix_train, label = wine$quality)
param <- list(max_depth = 2, eta = 1, silent = 1, nthread = 2,
              objective = "reg:linear")

wine_xgb_model <- xgb.train(param, data_train, nrounds = 50)

explainer_xgb <- explain(
  wine_xgb_model, 
  data = model_matrix_train, 
  y = wine$quality, 
  label = "xgboost"
)

pdp_old <- single_variable(
  explainer_xgb, 
  variable = "alcohol", 
  type = "pdp"
) # works

pdp_new <- partial_dependency(
  explainer_xgb, 
  variables = "alcohol"
) # fails
```

Results in message:
```
Error in 1:nrow(new_observation) : argument of length 0
In addition: Warning messages:
1: In profiles$`_label_` <- label : Coercing LHS to a list
2: In new_observation$`_yhat_` <- predict_function(x, new_observation) :
  Coercing LHS to a list
```
I managed to track this down to https://github.com/ModelOriented/ingredients/blob/53c7c3d46e932d54a46ae900027aaa2266b96ed7/R/ceteris_paribus.R#L129. This line converts `new_observation` to a list. 
Thus a couple of lines below,  calling `1:nrow(new_observation)` results in an error.

I don't know whether `partial_dependency` is intentionally meant to work only with explainer objects with data frames within them. However it seems that at least a check of the data type should be added and some more informative error message should be thrown.

Session info:
```
> sessionInfo()
R version 3.4.4 (2018-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.2 LTS

Matrix products: default
BLAS: /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=bg_BG.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=bg_BG.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=bg_BG.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=bg_BG.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ingredients_0.3.3 DALEX_0.4         xgboost_0.81.0.1  dplyr_0.8.0.1    

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.0        rstudioapi_0.9.0  magrittr_1.5      tidyselect_0.2.5  munsell_0.5.0     colorspace_1.4-0  lattice_0.20-35  
 [8] R6_2.4.0          rlang_0.3.1       plyr_1.8.4        tools_3.4.4       grid_3.4.4        data.table_1.12.0 gtable_0.2.0     
[15] lazyeval_0.2.1    yaml_2.2.0        assertthat_0.2.0  tibble_2.0.1      crayon_1.3.4      pdp_0.7.0         Matrix_1.2-12    
[22] gridExtra_2.3     purrr_0.3.0       ggplot2_3.1.0     glue_1.3.0        breakDown_0.1.6   stringi_1.3.1     compiler_3.4.4   
[29] pillar_1.3.1      scales_1.0.0      pkgconfig_2.0.2  
```

**Comment by pbiecek** [comment]
thanks, fixed in 8655a17

**Comment by pbiecek** [comment]
and now partial_dependency is much faster
